### PR TITLE
Fix RO for richChoiceView

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
Added back `dao.find()` partially reverting https://github.com/kgrgreer/foam3/pull/804. The code was required to initially set the value for the `fullObject_` on view render. Added forEach to sections to ensure the value can be in any section and still be read.

### CODE REVIEW

File: src/foam/u2/view/RichChoiceView.js [hunk 1]
```diff
@@ -617,7 +617,15 @@ foam.CLASS({
...
       name: 'onDataUpdate',
       code: function() {
-        if ( this.data === undefined ) this.clearSelection();
+        if ( this.data === undefined ) {
+           this.clearSelection();
+           return;
+        }
+        this.sections.forEach(section => {
+          section.dao.find(this.data).then(result => {
+            this.fullObject_ = result;
```
Comment (jinnnn9) [inline src/foam/u2/view/RichChoiceView.js]: Wouldn't `fullObject_` be just overridden to the result from the last find?

**Review by Nauna — APPROVED** [review]
will test once merged but makes sense

File: src/foam/u2/view/RichChoiceView.js (hunk 1 above)
Comment (sarthak-marwaha) [reply src/foam/u2/view/RichChoiceView.js]: Well there should be only one valid find since richChoiceView doesn't have multi-select support afaik

Comment (jinnnn9) [reply src/foam/u2/view/RichChoiceView.js]: Can we make sure that `fullObject_` will be set to the result from valid find though when you have multiple sections as find is done async?

Comment (sarthak-marwaha) [reply src/foam/u2/view/RichChoiceView.js]: We can check that `result != undefined` before setting `fullObject_ = result` but other than that, no way to make sure I guess

**Review by jinnnn9 — APPROVED** [review]

File: src/foam/u2/view/RichChoiceView.js [hunk 2]
```diff
@@ -617,7 +617,15 @@ foam.CLASS({
     {
       name: 'onDataUpdate',
       code: function() {
-        if ( this.data === undefined ) this.clearSelection();
+        if ( this.data === undefined ) {
```
Comment (Nauna) [inline src/foam/u2/view/RichChoiceView.js:620]: if ( ! this.data )

Comment (sarthak-marwaha) [reply src/foam/u2/view/RichChoiceView.js:620]: `this.data` can be 0 which will be falsy but might be a valid value hence the undefined

Comment (jinn9) [reply src/foam/u2/view/RichChoiceView.js:620]: this won't work since `data` is id property. Imagine when data is set to 0 which is falsy value

**Pull request merged**